client/my-sites/hosting/php-version-card/index.js

```diff
@@ -0,0 +1,220 @@
+					requestPhpVersion( siteId );
+				}
+				return [ [ requestId( siteId, method ), success ] ];
+			},
+			freshness: 0,
+		}
+	);
+};
+
+class PhpVersionCard extends React.Component {
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			selectedPhpVersion: null,
+		};
+	}
+
+	componentDidMount() {
+		requestPhpVersion( this.props.siteId );
+	}
+
+	changePhpVersion = event => {
+		const version = event.target.value;
+
+		this.setState( {
+			selectedPhpVersion: version,
+		} );
+	};
+
+	componentDidUpdate( prevProps ) {
+		const prevResponse = prevProps.updateResult;
+		const { updateResult, translate } = this.props;
+
+		const updateNoticeId = 'hosting-php-version';
+
+		if ( prevResponse && prevResponse !== 'failure' && updateResult === 'failure' ) {
+			this.props.errorNotice( translate( 'Failed to set PHP version.' ), {
+				id: updateNoticeId,
+			} );
+		}
+
+		if ( prevResponse && prevResponse !== 'success' && updateResult === 'success' ) {
+			this.props.successNotice(
+				translate( 'PHP version successfully set to %(version)s.', {
+					args: {
+						version: this.state.selectedPhpVersion,
+					},
+				} ),
+				{
+					id: updateNoticeId,
+					showDismiss: false,
+					duration: 5000,
+				}
+			);
+		}
+	}
+
+	getPhpVersions() {
+		return [
+			{
+				label: '7.2',
+				value: '7.2',
+			},
+			{
+				label: '7.3',
+				value: '7.3',
+			},
+			{
+				label: '7.4',
+				value: '7.4',
+			},
+		];
+	}
+
+	getContent() {
+		if ( this.props.loading ) {
+			return;
+		}
+
+		const { translate, siteId, updating, version } = this.props;
+		const { selectedPhpVersion } = this.state;
+
+		return (
+			<div>
+				<p>
+					{ translate(
+						'Your site is currently running PHP version {{strong}}%(version)s.{{/strong}}',
+						{
+							args: {
+								version,
+							},
+							components: {
+								strong: <strong />,
+							},
+						}
+					) }
+				</p>
+
+				<div className="php-version-card__radio">
+					<FormRadiosBar
+						checked={ selectedPhpVersion || version }
+						onChange={ this.changePhpVersion }
+						items={ this.getPhpVersions() }
+					/>
+				</div>
+				<Button
+					className="php-version-card__set-version"
+					onClick={ () => setPhpVersion( siteId, selectedPhpVersion ) }
+					busy={ updating }
+				>
+					<span>{ translate( 'Update PHP Version' ) }</span>
+				</Button>
+			</div>
+		);
+	}
+
+	render() {
+		const { translate, loading } = this.props;
+
+		return (
+			<Card className="php-version-card">
+				<MaterialIcon icon="dns" size={ 32 } />
+				<CardHeading>{ translate( 'PHP Version' ) }</CardHeading>
+				{ this.getContent() }
+				{ loading && <Spinner /> }
+			</Card>
+		);
+	}
+}
+
+export default connect(
+	state => {
+		const siteId = getSelectedSiteId( state );
+
+		const phpVersionGet = getHttpData( requestId( siteId, 'GET' ) );
+		const phpVersionUpdate = getHttpData( requestId( siteId, 'POST' ) );
+		const version = get( phpVersionGet, 'data', null );
```

Comment:
This usage is fine, but also fyi that can now use [optional chaining](https://github.com/tc39/proposal-optional-chaining/) in Calypso ✨ So we might update this to

```jsx
const version = phpVersionGet?.data ?? null;
```